### ISSUE TITLE
Configure inconsistency with Open MPI 4.0.3 and  UCX 1.8.0

### ISSUE THREAD

**Issue by cniethammer** [open-mpi/ompi#7612]
## Background information

I am building Open MPI 4.0.3 from the release tarball with the latest UCX 1.8.0 on a CentOS 8.0 based system with Infiniband (Mellanox ConnectX-6).

Open MPI was configured with
`./configure --with-ucx=<path_to_ucx_1.8.0_install_dir>`

UCX was build with default settings, no additional configure options except `--prefix`.

## Details of the problem

The configure step completes successfully and reports at the end
`Transports ... Open UCX: yes`.

However configure reports also
`--- MCA component btl:uct (m4 configuration macro)
checking for MCA component btl:uct compile mode... static
checking check uct version... yes
checking UCT version compatibility... UCT version not compatible - need UCX 1.7 or older
checking if MCA component btl:uct can compile... no`

This seem to be inconsistent for me.

I do not see any problems during make and with the final installation so far but maybe someone knowing the configure and UCX related code could check this.

**Comment by devreal** [comment]
Seeing the same message from configure in current `v4.0.x` branch.

**Comment by hppritcha** [comment]
This is exactly what should be expected.  The UCT BTL is only compatible with UCX up to version 1.7 so it doesn't get configured in.  You still get the UCX OSC and PML components.  The UCX OSC component is the preferred path for MPI RMA ops when using IB or ROCe interconnects.

**Comment by hppritcha** [comment]
@gpaulsen this is expected behavior.

**Comment by devreal** [comment]
@hppritcha if that is the case, the error message is misleading. It states `need UCX 1.7 or older`

**Comment by devreal** [comment]
@hppritcha my bad, of course "or older" refers to previous versions :/ the message is correct of course.

**Comment by bosilca** [comment]
UCX 1.8 works nicely with the UCT BTL. One just need to allow it with the following patch
```
diff --git a/opal/mca/btl/uct/configure.m4 b/opal/mca/btl/uct/configure.m4
index 61c7c8c..f125f3b 100644
--- a/opal/mca/btl/uct/configure.m4
+++ b/opal/mca/btl/uct/configure.m4
@@ -52,7 +52,7 @@ dnl
     fi

     max_allowed_uct_major=1
-    max_allowed_uct_minor=7
+    max_allowed_uct_minor=8
     if test "$btl_uct_happy" = "yes" && test "$enable_uct_version_check" != "no"; then
         AC_MSG_CHECKING([UCT version compatibility])
         OPAL_VAR_SCOPE_PUSH([CPPFLAGS_save])
```

**Comment by cniethammer** [comment]
@bosilca : Thanks, that was the info I was looking for! From the UCX 1.8.0 release notes I could not see changes related to the UCX API. So, it was also my impression that this is a simple extension of the allowed UCX version range.

**Comment by jsquyres** [comment]
@bosilca @cniethammer If you guys want that for v5.0, you might want to PR it soon -- branch is April 30.

**Comment by cniethammer** [comment]
@hjelmn : may I ask you to have a short look as I see that you had coped with this UCT code lastly in #7128

**Comment by devreal** [comment]
Is there a reason why the UCT btl is using a whitelist for UCX versions? Is it in anticipation of API breakage in the future? Or to turn the question around: is the UCT btl superseded by other components and thus safe to be dysfunctional starting with UCX 1.8.0 (or 1.9.0/2.0 if #7673 gets merged)? My concern that we will be at the exact same position with the next UCX release...

**Comment by cniethammer** [comment]
What I see in https://github.com/open-mpi/ompi/issues/7128#issuecomment-550198825 the used UCT API is not stable in this part.

**Comment by jsquyres** [comment]
@devreal @cniethammer Yes, exactly.  The UCT API is not guaranteed to be stable from release to release.  Before we whitelisted UCT versions, we were in a situation where new UCX releases broken Open MPI builds (because of the UCT BTL).  After significant debate in the community, we decided to put this whitelisting in place so that a new UCX release wouldn't break Open MPI's build -- especially when Open MPI is intentionally using a UCX API that is not guaranteed to be stable.

IIRC, there was supposed to be a configure CLI option to say "disregard the UCX whitelisting; go ahead and try to build the UCT BTL anyway", specifically to handle the case where a new UCX was released that *didn't* break the UCT BTL build.  Did that not happen?
